extensions/scheduler/runtime/src/main/java/io/quarkus/scheduler/runtime/SimpleScheduler.java

```diff
@@ -148,7 +148,10 @@ public void resume() {
     }
 
     SimpleTrigger createTrigger(String invokerClass, CronParser parser, Scheduled scheduled) {
-        String id = triggerNameSequence.getAndIncrement() + "_" + invokerClass;
+        String id = scheduled.identity().trim();
+        if (id.isEmpty()) {
+            id = triggerNameSequence.getAndIncrement() + "_" + invokerClass;
```

Comment:
We could probably get rid of `SimpleScheduler.triggerNameSequence` and `SimpleScheduler.config` fields as they are only used in the constructor, in `createTrigger()`...